src/algorithm/hnswlib/hnswalg.cpp

```diff
@@ -71,8 +71,19 @@ HierarchicalNSW::HierarchicalNSW(SpaceInterface* s,
     rev_size_ = 1.0 / mult_;
 }
 
+void
+HierarchicalNSW::reset() {
+    allocator_->Deallocate(element_levels_);
+    allocator_->Deallocate(reversed_level0_link_list_);
+    allocator_->Deallocate(reversed_link_lists_);
+    allocator_->Deallocate(molds_);
+    allocator_->Deallocate(link_lists_);
+}
+
 bool
 HierarchicalNSW::init_memory_space() {
+    // release the memory allocated by the init_memory_space function that was called earlier
+    reset();
```

Comment:
Will `init_memory_space` be called multiple times?

Reply:
In very rare cases, calling `init_memory_space` may result in a `bad_alloc` issue. Therefore, before initializing memory, it is necessary to release any previously allocated memory that may exist.

Reply:
OK, Also need to clean released memory to avoid double free